pkg/skaffold/deploy/deploy.go

```diff
@@ -34,6 +34,9 @@ type Deployer interface {
 	// Deploy should ensure that the build results are deployed to the Kubernetes
 	// cluster.
 	Deploy(context.Context, io.Writer, *build.BuildResult) (*Result, error)
+
+	// Cleanup deletes what was deployed by calling Deploy.
+	Cleanup(context.Context, io.Writer) error
```

Comment:
More a meta question outside this PR, but should we cleanup build resources also? Maybe skaffold runner could call cleanup from the builder and deployer.